lib/crawler.js

```diff
@@ -559,9 +563,17 @@ Crawler.prototype.discoverResources = function(resourceData, queueItem) {
     // Rough scan for URLs
     return crawler.discoverRegex
         .reduce(function(list, regex) {
+            var matches = resourceText.match(regex);
+
+            // If the RegExp doesn't contain a global flag, we expect to find the
+            // resources in the capture groups, so we strip away the match as a
+            // whole (ie. the first entry in the matches array).
+            if (regex.toString().split("/").slice(-1)[0].indexOf("g") === -1) {
```

Comment:
To avoid the entire split and slice part, there's https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/flags, but that's not supported in node.

But I just realized that to avoid this entire logic (which I admit might not be the most solid) I could actually just wrap the entire regexp in a non-capturing group! I'll fix that later today